### ISSUE TITLE
Select Subitem Within a Subitem

### ISSUE THREAD

**Issue by jonsaw** [go-pg/pg#280]
Is it possible to get a subitem's subitem? 

E.g., a _subitem within a subitem_:

```
type Country struct {
	ID   int
	Name string
}
type Author struct {
	ID        int
	Name      string
	CountryID int
	Country   *Country // has one relation
}
type Book struct {
	ID       int
	Title    string
	AuthorID int
	Author   *Author // has one relation
}

var books []Book
err := db.Model(&books).
	Column("book.*", "Author", "Author.Country"). // <- LEFT JOIN Author's Country
	Select()
```

**Comment by vmihailenco** [comment]
It should be possible, but is not implemented due to overlook. I hope to do it today.

**Comment by vmihailenco** [comment]
Fixed in v4.7.6

**Comment by matthewvalimaki** [comment]
Before I file a new issue wanted to confirm this here.

@vmihailenco should this as well work `Column("book.*", "Authors", "Authors.Country")`? For me `Column("book.*", "Authors")` works but adding `Authors.Country` fails with "reflect: Field of non-struct type". Authors is `Authors []*Author`.

**Comment by avitex** [comment]
I seem to be getting an issue like above

```go
type Group struct {
    // ...
    ParentId int
    Children []*Group `pg:",fk:Parent"`
    Members  []*User  `pg:",many2many:group_members"`
}
type User struct {
    // ...
    Groups []*Group `pg:",many2many:group_members"`
}
```

Attempting to query with columns like `pg.Model(&Group{}).Column("group.*", "Children").Select()` panics with:
`reflect: Field of non-struct type`

**Note:** This works when I am using `.First()`

**Comment by avitex** [comment]
Looking at the source, [typeByIndex, line 31](https://github.com/go-pg/pg/blob/d78aa3c8ec107aae735c627ac96b3abcc97baac2/orm/util.go#L31), it expects there to be a struct, when it's actually a ptr.
I don't know the structure of the library well enough to fix this. I did attempt a blind fix by jamming `indirectType` in, but the error just moved to somewhere else.

Hope this provides some sort of help @vmihailenco

**Comment by vmihailenco** [comment]
Thanks for report - I think I just fixed this in v4.8.0. Please try upgrade.

**Comment by matthewvalimaki** [comment]
@vmihailenco works for me now. Thank you for quick response!
